shell/platform/fuchsia/flutter/tests/fakes/scenic/fake_resources.h

```diff
@@ -0,0 +1,185 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_FUCHSIA_FLUTTER_TESTS_FAKES_SCENIC_FAKE_RESOURCES_H_
+#define FLUTTER_SHELL_PLATFORM_FUCHSIA_FLUTTER_TESTS_FAKES_SCENIC_FAKE_RESOURCES_H_
+
+#include <fuchsia/ui/gfx/cpp/fidl.h>
+
+#include <array>
+#include <memory>
+#include <set>
+#include <string>
+#include <variant>
+
+namespace flutter_runner::testing {
+
+using FakeResourceId = uint32_t;
+constexpr FakeResourceId kFakeResourceIdInvalid = 0u;
+
+struct FakeResource;
+
+struct FakeNodeInfo {
+  std::array<float, 4> rotation_quaternion = {0.f, 0.f, 0.f, 1.f};
+  std::array<float, 3> scale_vector = {1.f, 1.f, 1.f};
+  std::array<float, 3> translation_vector = {0.f, 0.f, 0.f};
+  std::array<float, 3> anchor_vector = {0.f, 0.f, 0.f};
+  fuchsia::ui::gfx::HitTestBehavior hit_test_behavior =
+      fuchsia::ui::gfx::HitTestBehavior::kDefault;
+  bool semantically_visible = true;
+
+  std::set<std::shared_ptr<FakeResource>> children;
+  std::shared_ptr<FakeResource> parent;
+
+  bool operator==(const FakeNodeInfo& other) const {
```

Comment:
Why not just give the nodes unique ID's?

Reply:
I split out the stuff involving Nodes, gonna land in another PR so we can review there